File: Java/ChatterBox/src/data/IUser.java
package data;

import java.awt.Color;

/*******************************************************************************
 * User interface.
 ******************************************************************************/
public interface IUser
{
    /***************************************************************************
     * Returns the name of this user. This name will not be seen by other users.
     * Eg. johndoe@localhost
     * 
     * @return
     **************************************************************************/
    public String getName();

    /***************************************************************************
     * Returns the displayname of this user. This is the name that will be seen
     * by other users.
     * 
     * @return
     **************************************************************************/
    public String getDisplayName();

    /***************************************************************************
     * Sets the display name of this user.
     * 
     * @param displayName
     **************************************************************************/
    public void setDisplayName( String displayName );

    /***************************************************************************
     * Returns the display color this user has selected.
     * 
     * @return
     **************************************************************************/
    public Color getDisplayColor();

    /***************************************************************************
     * Sets the display color for this user.
     * 
     * @param color
     **************************************************************************/
    public void setDisplayColor( Color color );

    /***************************************************************************
     * Returns true if this user is available, false otherwise.
     * 
     * @return
     **************************************************************************/
    public boolean isAvailable();

    /***************************************************************************
     * Sets the availability of this user.
     * 
     * @param available
     **************************************************************************/
    public void setAvailable( boolean available );
}

File: C#/ChatterBox/ChatterBox/IUser.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Drawing;

namespace ChatterBox
{
    interface IUser
    {
        /// <summary>
        /// Returns the machine name of this user.
        /// </summary>
        /// <returns></returns>
        string getName();

        /// <summary>
        /// Returns the displya name of this user.
        /// </summary>
        /// <returns></returns>
        string getDisplayName();

        /// <summary>
        /// Sets the display name of this user.
        /// </summary>
        /// <param name="displayName"></param>
        void setDisplayName( string displayName );

        /// <summary>
        /// Returns true if this user is available, false otherwise.
        /// </summary>
        /// <returns></returns>
        bool isAvailable();

        /// <summary>
        /// Sets the availability of this user.
        /// </summary>
        /// <param name="available"></param>
        void setAvailable( bool available );

        /// <summary>
        /// Returns this user's preferred color.
        /// </summary>
        /// <returns></returns>
        Color getPreferredColor();

        /// <summary>
        /// Sets this user's preferred color.
        /// </summary>
        /// <param name="color"></param>
        void setPreferredColor( Color color );
    }
}
